src/emulator/commandUtils.ts

```diff
@@ -156,6 +156,20 @@ export function parseInspectionPort(options: any): number {
   return parsed;
 }
 
+export function shutdownWhenKilled(): Promise<void> {
+  return new Promise((res, rej) => {
+    process.on("SIGINT", async () => {
+      try {
+        await controller.cleanShutdown();
+        res();
+        process.exit(0);
```

Comment:
Is `process.exit` necessary here? Wouldn't commands exit by themselves normally if `res()` is called?

If not needed, I'd advice dropping this line so that each command can do additional cleanup like

```js
await killSignalPromise;
await additionalCleanup();  // <-- this won't run if `process.exit(0)` was called.
```

Reply:
I need to be able to exit if Ctrl+C is pressed at any time, even if `controller.startAll()` is still in progress.